CMakeLists.txt

```diff
@@ -32,6 +32,28 @@ if (CYGWIN)
   set(CMAKE_FIND_LIBRARY_SUFFIXES ".dll")
 endif()
 
+find_package(Git REQUIRED)
+add_custom_command(
+  OUTPUT "${CMAKE_BINARY_DIR}/version_gen.h.tmp"
+  COMMAND "${CMAKE_COMMAND}" -E echo_append "#define ALIVE_VERSION_MACRO " > "${CMAKE_BINARY_DIR}/version_gen.h.tmp"
+  COMMAND "${GIT_EXECUTABLE}" describe --tags --dirty >> "${CMAKE_BINARY_DIR}/version_gen.h.tmp"
+  WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
+  VERBATIM
+)
+add_custom_command(
+  DEPENDS "${CMAKE_BINARY_DIR}/version_gen.h.tmp"
+  OUTPUT "${CMAKE_BINARY_DIR}/version_gen.h"
+  COMMAND "${CMAKE_COMMAND}" -E copy_if_different "${CMAKE_BINARY_DIR}/version_gen.h.tmp" "${CMAKE_BINARY_DIR}/version_gen.h"
+  COMMAND "${CMAKE_COMMAND}" -E remove -f "${CMAKE_BINARY_DIR}/version_gen.h.tmp"
```

Comment:
This indirection is needed so that the version is auto-updated each time it can be updated,
without the need to re-running cmake.
This does of course mean that second `ninja` in `ninja && ninja` will not be a NOP.